File: two-sums/two-sums.py
class Solution:
    def twoSum(self, nums, target):
        nums_len = len(nums)
        hash_map = {}
        result = []
        
        for index in range(nums_len):
            difference = target - nums[index]
            if(difference in hash_map):
                result.append(index)
                result.append(hash_map[difference])
                return result
            
            hash_map[nums[index]] = index
        
        return result

File: two-sums/two-sums.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
	const hashMap = {};
	const result = [];
	let difference;

	for (let index = 0; index < nums.length; index++) {
		difference = target - nums[index];
		if (difference in hashMap) {
			result[0] = index;
			result[1] = hashMap[difference];
			return result;
		}

		hashMap[nums[index]] = index;
	}

	return result;
};
